### PR TITLE
homework 28.1 solution

### PR DESCRIPTION
**Pull request by tkhambir**

### CODE REVIEW

**Review by TimoxaV — APPROVED** [review]

File: src/main/java/com/internet/shop/controllers/LoginController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("pwd");
+        try {
+            User user = authenticationService.login(login, password);
+            HttpSession session = req.getSession();
+            session.setAttribute("userId", user.getId());
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/LoginController.java]: let's make `userId` a constant

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        return userService.findByLogin(login)
+                .filter(u -> u.getPassword().equals(password))
+                .orElseThrow(() -> new AuthenticationException("Incorrect username or password."));
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java:18]: looks nice!

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthenticationFilter implements Filter {
+
+    private static Injector injector = Injector.getInstance("com.internet.shop");
+    private final UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]: redundant empty line

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            resp.sendRedirect("/login");
+            return;
+        }
+
+        chain.doFilter(req, resp);
+    }
+
+    @Override
+    public void destroy() {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]: same

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse resp = (HttpServletResponse) response;
+
+        String url = req.getServletPath();
+        if (url.equals("/login") || url.equals("/users/registration")) {
+            chain.doFilter(req, resp);
+            return;
+        }
+
+        Long userId = (Long) req.getSession().getAttribute("userId");
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]: let's make `userId` a constant

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**